### PR TITLE
buildRustPackage: fix cargo flag for release/debug build

### PR DESCRIPTION
**Pull request by adelbertc**
###### Motivation for this change
Previous behavior did not actually allow for "debug" build as it would
try to pass --debug to cargo, which is not a valid flag.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [X] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Assured whether relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by c0bw3b** [comment]
Related: #56540
cc @Mic92 @Ericson2314 @illegalprime for review

**Comment by adelbertc** [comment]
Ping?

File: pkgs/build-support/rust/default.nix [hunk 1]
```diff
@@ -103,7 +105,7 @@ in stdenv.mkDerivation (args // {
       "CC_${stdenv.hostPlatform.config}"="${ccForHost}" \
       "CXX_${stdenv.hostPlatform.config}"="${cxxForHost}" \
       cargo build \
-        --${buildType} \
+        ${releaseFlag} \
```
Comment (Mic92) [inline pkgs/build-support/rust/default.nix]:
```suggestion
        ${optionalString (buildType == "release") "--release"} \
```

Comment (Ericson2314) [reply pkgs/build-support/rust/default.nix]: Yeah that is better.

**Pull request merged**